File: day2/numSum.py
# write a program that prompts the user to enter
# a positive integer, then display the sum of it's digits

user_input = input("Enter a positive integer: ")
while not user_input.isdigit():
    user_input = input("Please enter a valid positive whole number: ")

user_ints = [int(x) for x in user_input]
print("The sum is", sum(user_ints))

File: day2/num_sum.rs
use std::io;

fn main() {
    loop {
        println!("\nPlease enter a positive whole number:");

        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read input");

        let trimmed = input.trim();

        if trimmed.chars().all(|c| c.is_ascii_digit()) {
            let sum: u32 = trimmed.chars()
                .map(|c| c.to_digit(10).unwrap())
                .sum();

            println!("The sum of the digits is: {}", sum);
            break;
        } else {
            println!("Please only enter a positive whole number.");
        }
    }
}
